src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs

```diff
@@ -0,0 +1,803 @@
+
+                var schemeAppCredentials = new SchemeApplicationCredentials()
+                {
+                    Scheme = ApplicationCredentialsScheme.AdminSetsCredentials,
+                    UserNameTemplate = new ApplicationCredentialsUsernameTemplate()
+                    {
+                        Template = "${source.login}",
+                        Type = "BUILT_IN",
+                    },
+                };
+
+                retrieved.Credentials = schemeAppCredentials;
+                retrieved = await client.Applications.UpdateApplicationAsync(retrieved, retrieved.Id);
+
+                retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.AdminSetsCredentials);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task UpdateSWAApplicationSetUserEditablePassword()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new SwaApplication
+            {
+                Name = "template_swa",
+                Label = "Sample Plugin App",
+                SignOnMode = ApplicationSignOnMode.BrowserPlugin,
+                Settings = new SwaApplicationSettings
+                {
+                    App = new SwaApplicationSettingsApplication
+                    {
+                        ButtonField = "btn-login",
+                        PasswordField = "txtbox-password",
+                        UsernameField = "txtbox-username",
+                        Url = "https://example.com/login.html",
+                        LoginUrlRegex = "^https://example.com/login.html",
+                    },
+                },
+            });
+
+
+            try
+            {
+                var retrieved = await client.Applications.GetApplicationAsync(createdApp.Id);
+
+                // Checking defaults
+                var retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.EditUsernameAndPassword);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+
+                var schemeAppCredentials = new SchemeApplicationCredentials()
+                {
+                    Scheme = ApplicationCredentialsScheme.EditPasswordOnly,
```

Comment:
Does this work if you change the original `retrievedCredentials` object, instead of creating a new `SchemeApplicationCredentials`?